sympy/stats/tests/test_rv.py

```diff
@@ -153,6 +153,15 @@ def test_random_symbol_no_pspace():
     x = RandomSymbol(Symbol('x'))
     assert x.pspace == PSpace()
 
+def test_iid():
+    raises(TypeError, lambda: iid("I", Normal("N", 0, 1)))
+    raises(ValueError, lambda: iid("I", Normal("N", 0, 1), -1))
+
+    I = iid("I", Normal("N", 0, 1), 5)
+    assert P(I[1] > 0) == P(I[2] > 0) == Rational(1, 2)
+    assert P(I[3] > 0, I[1] < 0) == P(I[3] > 0)
+    assert E(I[1] * I[2] * I[3]) == E(I[1]) * E(I[2]) * E(I[3])
```

Comment:
Can we model the same queries using joint distributions?

Reply:
As of now, computing the probability for joint distributions is not implemented. Further - 
``` python
>>> a1, a2 = symbols('a1 a2', positive=True)
>>> B = MultivariateBeta('B', [a1, a2])
>>> C = MultivariateBeta('C', [a1, a2])
>>> E(B[0]*C[0])
B[0] * C[0]
```
Is this expected? All the results are the same when I use IID

Reply:
Can you find any use case where we are compelled to model our requirement using IIDs and the same cannot be done by JointDistributions?

Reply:
I think there are no cases where we would be forced to use IIDs. Thinking about it now, I do feel like this is not really a useful feature to add. But how do we do this with JointDistributions?

Reply:
Joint distributions in `joint_rv_types.py` only allows for "rectangular" supports (sets). The joint distribution of order statistics needs to have a triangular set and I don't think this is possible in the joint_rv.